### PR TITLE
Replace backticks (``) with $() for command substitution

### PR DESCRIPTION
**Pull request by anbj**
Replace the legacy syntax of backticks (``) with the preferred $() for command substitution.

### CODE REVIEW

**Comment by PaulWessel** [comment]
I had to learn about the legacy/obsoleteness of backticks first [here](https://stackoverflow.com/questions/9449778/what-is-the-benefit-of-using-instead-of-backticks-in-shell-scripts).  I suppose this is progress.  Would you agree, @GenericMappingTools/core ?

**Comment by anbj** [comment]
See also http://mywiki.wooledge.org/BashFAQ/082.

**Comment by seisman** [comment]
Looks good to me if all the tests still pass, but csh may not support `$()` sytnax.

**Comment by PaulWessel** [comment]
Only csh script we have is gmt_aliases.csh and it has
`set gmt_modules = (`gmt --show-classic`)`
which fails if we do
```
set gmt_modules = $(gmt --show-classic)
Illegal variable name.
```
So we can leave that as is.  No other csh syntax is shown or used anywhere.  We do not want to encourage csh use anyway.

**Comment by PaulWessel** [comment]
Hi @anbj, did you run all the tests to see if there were any new failures?

**Comment by anbj** [comment]
No, sorry - not yet. I can run the tests tomorrow. I can also revert the changes for ``gmt_aliases.csh``.

**Comment by seisman** [comment]
I just ran all tests locally. There are 4 failing tests:
```
	146 - anim01/anim01.sh (Failed)
	147 - anim02/anim02.sh (Failed)
	151 - anim06/anim06.sh (Failed)
	287 - gmtregress/draper.sh (Failed)
```

File: doc/examples/anim01/anim01.sh [hunk 1]
```diff
@@ -25,7 +25,7 @@ EOF
 cat << EOF > main.sh
 gmt begin
 #	Plot smooth blue curve and dark red dots at all angle steps so far
-	last=\`gmt math -Q \${MOVIE_FRAME} 10 MUL =\`
+	last=\$(gmt math -Q \${MOVIE_FRAME} 10 MUL =\)
```
Comment (seisman) [inline doc/examples/anim01/anim01.sh]:
```suggestion
	last=\$(gmt math -Q \${MOVIE_FRAME} 10 MUL =)
```

File: doc/examples/anim02/anim02.sh [hunk 2]
```diff
@@ -22,7 +22,7 @@ EOF
 # 2. Set up the main frame script
 cat << EOF > main.sh
 gmt begin
-	width=\`gmt math -Q \${MOVIE_WIDTH} 0.5i SUB =\`
+	width=\$(gmt math -Q \${MOVIE_WIDTH} 0.5i SUB =\)
```
Comment (seisman) [inline doc/examples/anim02/anim02.sh]:
```suggestion
	width=\$(gmt math -Q \${MOVIE_WIDTH} 0.5i SUB =)
```

File: doc/examples/anim06/anim06.sh [hunk 3]
```diff
@@ -46,27 +46,27 @@ gmt begin
 	# Plot the shifted chirp
 	gmt plot \$R \$J chirp_shifted.txt -W1p,red -X0.2i -Y0.3i
 	# Compute index of most recent sample number
-	last_sample=\`gmt math -Q \${MOVIE_FRAME} \$rate 2 DIV DIV FLOOR RINT =\`
+	last_sample=\$(gmt math -Q \${MOVIE_FRAME} \$rate 2 DIV DIV FLOOR RINT =\)
```
Comment (seisman) [inline doc/examples/anim06/anim06.sh]:
```suggestion
	last_sample=\$(gmt math -Q \${MOVIE_FRAME} \$rate 2 DIV DIV FLOOR RINT =)
```

File: doc/examples/anim06/anim06.sh [hunk 4]
```diff
@@ -46,27 +46,27 @@ gmt begin
...
 	# Extract all the old samples before the present
 	gmt convert chirp_samples.txt -Z:\$last_sample > tmp.txt
 	if [ -s tmp.txt ]; then
 		gmt math tmp.txt -C0 \${MOVIE_COL1} SUB = samples.txt
 		gmt plot -Sc0.3c -Gblue samples.txt
 	fi
 	# Take a new sample every 12 frames = 0.5 seconds
-	take_sample=\`gmt math -Q \${MOVIE_FRAME} \$rate 2 DIV MOD 0 EQ =\`
+	take_sample=\$(gmt math -Q \${MOVIE_FRAME} \$rate 2 DIV MOD 0 EQ =\)
```
Comment (seisman) [inline doc/examples/anim06/anim06.sh]:
```suggestion
	take_sample=\$(gmt math -Q \${MOVIE_FRAME} \$rate 2 DIV MOD 0 EQ =)
```

File: doc/examples/anim06/anim06.sh [hunk 5]
```diff
@@ -46,27 +46,27 @@ gmt begin
...
 	if [ \${MOVIE_FRAME} -gt 12 ]; then	# Interpolating up to most recent sample
 		gmt sample1d samples.txt -I0.001 > resampled.txt
 		gmt plot -W2.5p,blue resampled.txt
 	fi
 	if [ \$take_sample -eq 1 ]; then	# Take and plot sample at zero time
-		y=\`gmt math -Q \${MOVIE_COL1} 2 POW 2 DIV 60 DIV \$f MUL 2 MUL PI MUL COS =\`
+		y=\$(gmt math -Q \${MOVIE_COL1} 2 POW 2 DIV 60 DIV \$f MUL 2 MUL PI MUL COS =\)
```
Comment (seisman) [inline doc/examples/anim06/anim06.sh]:
```suggestion
		y=\$(gmt math -Q \${MOVIE_COL1} 2 POW 2 DIV 60 DIV \$f MUL 2 MUL PI MUL COS =)
```

File: doc/examples/anim06/anim06.sh [hunk 6]
```diff
@@ -46,27 +46,27 @@ gmt begin
...
 		echo 0 \$y | gmt plot -Sc0.5c -Gred
 	fi
 	# Add time counter in upper left corner
 	printf "%4.1f 2 t = %6.3f s\n" -7.5 \${MOVIE_COL1} | gmt text -F+f18p,Helvetica-Bold+jTL -Dj0.1i/0.1i
 	# Add cycles counter in upper right corner
-	fnow=\`gmt math -Q \${MOVIE_COL1} 60 DIV \$f MUL =\`
+	fnow=\$(gmt math -Q \${MOVIE_COL1} 60 DIV \$f MUL =\)
```
Comment (seisman) [inline doc/examples/anim06/anim06.sh]:
```suggestion
	fnow=\$(gmt math -Q \${MOVIE_COL1} 60 DIV \$f MUL =)
```

**Comment by anbj** [comment]
Great @seisman , thanks for catching these!

File: test/gmtregress/draper.sh [hunk 7]
```diff
@@ -32,7 +32,7 @@ cat << EOF > draper.txt
 28.6	11.08
 EOF
 # First plot data and basic LS fit with equation
-txt=`gmt regress -Ey -N2 -Fxm -T25/80/2+n draper.txt -T0 | awk '{printf "85 13 @!y\\\\303 = %.4f %.4f x\n", $17, $15}'`
+txt=$(gmt regress -Ey -N2 -Fxm -T25/80/2+n draper.txt -T0 | awk '{printf "85 13 @!y\\\\303 = %.4f %.4f x\n", $17, $15}')
```
Comment (seisman) [inline test/gmtregress/draper.sh]:
```suggestion
txt=$(gmt regress -Ey -N2 -Fxm -T25/80/2+n draper.txt -T0 | awk '{printf "85 13 @!y\\303 = %.4f %.4f x\n", $17, $15}')
```
This will fix the draper.sh failure.

File: share/tools/gmt_aliases.csh [hunk 8]
```diff
@@ -16,7 +16,7 @@ if ($? == 1) then
   exit 1
 endif
 
-set gmt_modules = (`gmt --show-classic`)
+set gmt_modules = ($(gmt --show-classic))
```
Comment (seisman) [inline share/tools/gmt_aliases.csh]: Now all tests pass for me. 

Please revert the changes to `gmt_aliases.csh`, then this PR is good to merge.

**Comment by PaulWessel** [comment]
Thanks for this work @anbj !
